apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -1834,7 +1826,14 @@ mod test_finalize_block {
                 votes: votes.clone(),
                 ..Default::default()
             };
+            // merkle tree root before finalize_block
+            let root_pre = shell.shell.wl_storage.storage.block.tree.root();
+
             let _events = shell.finalize_block(req).unwrap();
+
+            // the merkle tree root should not change after finalize_block
+            let root_post = shell.shell.wl_storage.storage.block.tree.root();
+            assert_eq!(root_pre.0, root_post.0);
```

Comment:
when I remove the fixes and run the tests they do not yet catch the issue - I think we need to add some txs in here

Reply:
added in f30e158585cba829c9055cd218159e2e89d0fafe